packages/shared-state-dnsmasq_leases/files/usr/bin/shared-state-publish_dnsmasq_leases

```diff
@@ -45,9 +45,14 @@ function add_dhcp_leases_from_files()
 
 	for line in io.lines("/tmp/dhcp.leases") do
 		_, client_mac, client_ip, client_hostname, client_id = unpack(line:split(" "))
-		leasesTable[client_ip] = { hostname=client_hostname, mac=client_mac }
-
-		if type(client_hostname) == "string" and client_hostname:len() > 1 then
+		-- client_ip will be nil in duid lines
+		-- https://dnsmasq-discuss.thekelleys.org.narkive.com/ia9YNLmE/dnsmasq-leases-file-format-specification
+		if client_ip and type(client_hostname) == "string" and client_hostname:len() > 1 then
```

Comment:
The dhcp.leases file contains lines with hostname `*`, i.e.
```
1675107518 12:34:65:3d:ce:2b 10.51.147.128 * 01:12:34:65:3d:ce:2b
1675106752 ab:cd:15:8a:62:93 10.51.147.146 * 01:ab:cd:15:8a:62:93
```
It is correct that we don't want to add `*` into the `hostTable`. But I think we still want to add them into the `leasesTable`, which will not happen with this arrangement of if-statements.

Reply:
Ah ok, it makes sense, thanks! I am going to correct this today.